src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceRecallTest.kt

```diff
@@ -0,0 +1,868 @@
+import uk.gov.justice.hmpps.prison.service.receiveandtransfer.TrustAccountService
+import uk.gov.justice.hmpps.prison.util.builders.OffenderBookingBuilder
+import uk.gov.justice.hmpps.prison.util.builders.OffenderBuilder
+import uk.gov.justice.hmpps.prison.util.builders.getBedAssignments
+import uk.gov.justice.hmpps.prison.util.builders.getCaseNotes
+import uk.gov.justice.hmpps.prison.util.builders.getCurrentIEP
+import uk.gov.justice.hmpps.prison.util.builders.getMovements
+import uk.gov.justice.hmpps.prison.util.builders.transferOut
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+
+@WithMockUser
+class OffenderResourceRecallTest : ResourceTest() {
+
+  @MockBean
+  private lateinit var trustAccountService: TrustAccountService
+
+  @Nested
+  @DisplayName("POST /offenders/{offenderNo}/recall")
+  inner class RecallOffender {
+
+    @Nested
+    @DisplayName("when recall is rejected")
+    inner class Failure {
+      @Test
+      internal fun `404 when offender not found`() {
+        val offenderNo = "Z9999ZZ"
+
+        // Given offender does not exist
+        webTestClient.get()
+          .uri("/api/offenders/{offenderNo}", offenderNo)
+          .headers(
+            setAuthorisation(
+              listOf("ROLE_SYSTEM_USER")
+            )
+          )
+          .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+          .accept(MediaType.APPLICATION_JSON)
+          .exchange()
+          .expectStatus().isNotFound
+
+        // when recall is requested
+        webTestClient.put()
+          .uri("/api/offenders/{offenderNo}/recall", offenderNo)
+          .headers(
+            setAuthorisation(
+              listOf("ROLE_TRANSFER_PRISONER")
+            )
+          )
+          .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+          .bodyValue(
+            """
+            {
+               "prisonId": "SYI", 
+               "recallTime": "2020-01-01T12:00:00",
+               "fromLocationId": "COURT1", 
+               "movementReasonCode": "24", 
+               "youthOffender": "true", 
+               "imprisonmentStatus": "CUR_ORA", 
+               "cellLocation": "SYI-A-1-1"     
+            }
+            """.trimIndent()
+          )
+          .accept(MediaType.APPLICATION_JSON)
+          .exchange()
+          .expectStatus().isNotFound
+      }
+
+      @Test
+      internal fun `404 when offender has no bookings`() {
+        val offenderNo = createPrisonerWithNoBooking()
+
+        // Given offender has no bookings
+        webTestClient.get()
+          .uri("/api/offenders/{offenderNo}", offenderNo)
+          .headers(
+            setAuthorisation(
+              listOf("ROLE_SYSTEM_USER")
+            )
+          )
+          .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
+          .accept(MediaType.APPLICATION_JSON)
+          .exchange()
+          .expectStatus().isOk
```

Comment:
Should this assert the result is empty?

Reply:
I guess you mean assert that the test offender we just created has no bookings? If so, yes, sounds like a good idea. Next PR